common/models/change.js

```diff
@@ -138,11 +144,12 @@ module.exports = function(Change) {
 
   Change.findOrCreateChange = function(modelName, modelId, callback) {
     assert(loopback.findModel(modelName), modelName + ' does not exist');
+    callback = callback || utils.createPromiseCallback();
     var id = this.idForModel(modelName, modelId);
     var Change = this;
 
     this.findById(id, function(err, change) {
-      if (err) return callback(err);
+      if (err) callback(err);
```

Comment:
The return is important here! If you don't return from the function, then the code below is run even when there was an error.

Please revert.